src/main/java/graphql/nadel/result/ResultComplexityAggregator.java

```diff
@@ -0,0 +1,46 @@
+package graphql.nadel.result;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class ResultComplexityAggregator {
+    private Integer totalNodeCount;
+    private LinkedHashMap<String, Integer> serviceNodeCountsMap;
+
+
+    public ResultComplexityAggregator(Integer totalNodeCount, LinkedHashMap<String, Integer> serviceNodeCountsMap) {
+        this.totalNodeCount = totalNodeCount;
+        this.serviceNodeCountsMap = serviceNodeCountsMap;
+    }
+
+    public Integer getTotalNodeCount() {
+        return totalNodeCount;
+    }
+
+    public LinkedHashMap<String, Integer> getServiceNodeCountsMap() {
```

Comment:
geenric types always

`Map<String, Integer> getServiceNodeCountsMap `

and the "Map" part in "getServiceNodeCountsMap" is kinda redundant - you know its a map from return type.  I would just call it `getServiceNodeCounts`